src/entries/demo/Demo.js

```diff
@@ -16,38 +18,77 @@ const hash = window.location.hash
 @styles
 export class Demo extends React.PureComponent {
   state = {
-    definitionUrl
+    definitionUrl,
+    showDialog: false,
+    definition: '',
+    useDefinition: false
   }
 
-  useUrlInput = () => {
-    this.setState({ definitionUrl: this.urlInput.value })
-  }
+  useUrlInput = () => this.setState({ definitionUrl: this.urlInput.value })
+  setUrlInput = (input) => { this.urlInput = input }
+  updateDefinition = (event) => this.setState({ useDefinition: false, definition: event.target.value })
+  toggleDialog = () => this.setState({ showDialog: !this.state.showDialog })
 
-  setUrlInput = (input) => {
-    this.urlInput = input
+  renderInputDefinition = () => {
+    this.setState({ useDefinition: true, definitionUrl: '', showDialog: false })
   }
 
   render () {
```

Comment:
Nah, let's wait until the demo is pulled away... need to try and keep velocity up while we get features in.